### PR TITLE
fix gdn padding mask prob

### PR DESCRIPTION
**Pull request by FeixLiu**
fix gdn padding mask prob

### CODE REVIEW

File: src/paddlefleet/transformer/gated_delta_net.py [hunk 1]
```diff
@@ -255,6 +260,63 @@ def reset_parameters(self):
...
+    ) -> paddle.Tensor | None:
+        """Derive a padding mask (1.0=valid, 0.0=padding) for GDN."""
+        is_sp = self.config.sequence_parallel and self.sp_size > 1
+
+        if attention_mask is not None and attention_mask.ndim == 2:
+            full_seq = attention_mask.shape[-1]
+            if is_sp:
+                # attention_mask is [b, full_s], slice to local chunk
+                seq_len_local = seq_len // self.sp_size
+                tp_rank = get_pg_rank(self.pg_collection.tp)
```
[the remark this answers is not in the document]
Comment (FeixLiu) [reply src/paddlefleet/transformer/gated_delta_net.py]: https://github.com/PaddlePaddle/PaddleFleet/pull/994/changes#diff-2db4fd44369169a983153d71bce3057c0d51a37ff365916ba68b527e47f90919R352 ？

File: tests/single_card_tests/transformer/test_gdn_padding_mask.py [hunk 2]
```diff
@@ -0,0 +1,361 @@
+# Copyright (c) 2026 PaddlePaddle Authors. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (FeixLiu) [reply tests/single_card_tests/transformer/test_gdn_padding_mask.py:1]: 你要不要看看现在是哪年哪月？

**Review by From00 — APPROVED** [review]
LGTM
